server/src/services/asset.service.ts

```diff
@@ -323,7 +323,21 @@ export class AssetService {
     }
 
     for (const id of ids) {
-      await this.updateMetadata({ id, dateTimeOriginal, latitude, longitude });
+      const asset = await this.assetRepository.getById(id);
+      const oldCreatedAtDate = asset?.fileCreatedAt && DateTime.fromJSDate(asset.fileCreatedAt);
+      let newDateTimeString = dateTimeOriginal;
+      if (dateTimeOriginal && keepTimeUnchanged && oldCreatedAtDate) {
+        let newDateTime = DateTime.fromISO(dateTimeOriginal);
+
+        newDateTime = newDateTime.set({
+          hour: oldCreatedAtDate.hour,
+          minute: oldCreatedAtDate?.minute,
+          second: oldCreatedAtDate.second,
+        });
+        const newDateTimeStringWithNull = newDateTime?.toISO();
+        newDateTimeString = newDateTimeStringWithNull === null ? undefined : newDateTimeStringWithNull;
+      }
```

Comment:
This is inside a for loop, which might be called with 10k assets. Loading an asset from the database one at a time is going to be a problem. In hindsight, calling updateMetadata one at a time, which eventually calls upsertExif might not be the best either. I would probably recommend finding a more optimized solution. Ideally we move this logic into updateMetadata itself, and change that method to accept a list of ids to update. Only if the keepTimeUnchanged property is set, we can use the assetRepo.getByIds methods to load them in a single query.

Reply:
We could also move the "keepTimeUnchanged" logic inside the SIDECAR_WRITE job or create a new one for that case.

The problem I see with changing multiple assets in one Job is the error handling.
What should happen if one of the metadata updates fail?

And maybe one side question:
is `await this.jobRepository.queue` waiting for the job to complete, or is it fire and forget. And what would happen if the SIDECAR_WRITE fails and the 

`await this.assetRepository.updateAll(ids, options); // asset.services.ts L 343 `

still is done.

Reply:
.queue is waiting for the job to be added to the queue. It is fine if the job takes longer to execute since it happens in the background. 

Currently the bulk update (and single update) method optimistically update the database so the UI can refresh immediately and see the changes, but then _persist_ the changes to a sidecar file so future metadata extraction doesn't erase/override the changes.

It is important that the initial implementation is fast, since the caller is waiting. Updating every asset individually makes that a little bit more difficult, especially if the asset needs to be read ahead of time.

Reply:
The options are to not do any update immediately, but defer it to the job. The user might be confused as the time isn't updated and the assets still appear in the wrong place for some time. 

Or, update each asset one at a time, but optimize the loading to be in a single database request.